googletest/include/gtest/internal/gtest-internal.h

```diff
@@ -1363,19 +1367,27 @@ constexpr bool InstantiateTypedTestCase_P_IsDeprecated() { return true; }
     GTEST_CONCAT_TOKEN_(gtest_label_testthrow_, __LINE__): \
       fail(gtest_msg.value)
 
+// We need to declare the ConstCharPtr object in the scope of the if. It
+// can't go into the body of the if because then we cannot access it in the
+// else. It also can't go before the if block because than we have to make up
+// some very unique name (line number is not enough).
```

Comment:
> It also can't go before the if block because than we have to make up some very unique name (line number is not enough).

This is not completely accurate - there is much deeper problem:
We cannot go before the if block because `ASSERT_NO_THROW(...)` should expand to a single statement, so it can be used as:
```cpp
if (some_condition)
    ASSORT_NO_THROW(some_expression) << "Additional error message";
```